extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTools.java

```diff
@@ -1417,15 +1619,67 @@ private static KnownLocationKeyFinder<ParquetTableLocationKey> toKnownKeys(
     /**
      * Reads in a table using metadata files found in the supplied directory.
      *
-     * @param directory The source of {@link ParquetTableLocationKey location keys} to include
+     * @param directory the path for the root directory to search for .parquet files
      * @param readInstructions Instructions for customizations while reading
      * @return The table
+     * @deprecated Use {@link #readTable(String, ParquetInstructions)} instead with layout provided as
+     *             {@link ParquetFileLayout#METADATA_PARTITIONED} using
+     *             {@link ParquetInstructions.Builder#setFileLayout}.
      */
+    @Deprecated
     public static Table readPartitionedTableWithMetadata(
             @NotNull final File directory,
             @NotNull final ParquetInstructions readInstructions) {
+        return readPartitionedTableWithMetadata(convertToURI(directory, true), readInstructions);
+    }
+
+    /**
+     * Reads in a table using metadata files found in the supplied directory.
+     *
+     * @param directory the path or URI for the root directory to search for .parquet files
+     * @param readInstructions Instructions for customizations while reading
+     * @return The table
+     * @deprecated Use {@link #readTable(String, ParquetInstructions)} instead with layout provided as
+     *             {@link ParquetFileLayout#METADATA_PARTITIONED} using
+     *             {@link ParquetInstructions.Builder#setFileLayout}.
+     */
+    @Deprecated
+    public static Table readPartitionedTableWithMetadata(
+            @NotNull final String directory,
+            @NotNull final ParquetInstructions readInstructions) {
+        return readPartitionedTableWithMetadata(convertToURI(directory, true), readInstructions);
+    }
+
+    private static Table readPartitionedTableWithMetadata(
+            @NotNull final URI sourceURI,
+            @NotNull final ParquetInstructions readInstructions) {
+        if (!FILE_URI_SCHEME.equals(sourceURI.getScheme())) {
+            throw new UnsupportedOperationException("Reading metadata files from non local storage is not supported");
+        }
+        verifyFileLayout(readInstructions, ParquetFileLayout.METADATA_PARTITIONED);
+        if (readInstructions.getTableDefinition().isPresent()) {
+            throw new UnsupportedOperationException("Detected table definition inside read instructions, reading " +
+                    "metadata files with custom table definition is currently not supported");
+        }
```

Comment:
This kind of suggests that definition should be a property of the layout. On the other hand, we could support table definition to mask out some columns or infer null for others. Overlapping columns must have consistent types.

Kind of like having a "split table function" as a property of write layouts.